lib/assist/assist.go

```diff
@@ -183,157 +182,38 @@ func getAssistantClient(ctx context.Context, proxyClient auth.ClientI,
 	return ai.NewClient(apiKey), nil
 }
 
-// InsertAssistantMessage inserts a message from the user into the conversation.
-// Message is saved in the DB.
-func (c *Chat) InsertAssistantMessage(ctx context.Context, msgType MessageType, msgPayload string,
-) error {
-	// write a user message to both an in-memory chain and persistent storage
-	c.chat.Insert(kindToRole(msgType), msgPayload)
+// ProcessComplete processes the completion request and returns the number of tokens used.
+func (c *Chat) ProcessComplete(ctx context.Context,
+	onMessage func(kind MessageType, payload []byte, createdTime time.Time) error, userInput string,
+) (*model.TokensUsed, error) {
+	var tokensUsed *model.TokensUsed
+
+	// query the assistant and fetch an answer
+	message, err := c.chat.Complete(ctx, userInput)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 
+	// write the user message to persistent storage and the chat structure
+	c.chat.Insert(openai.ChatMessageRoleUser, userInput)
 	if err := c.authClient.CreateAssistantMessage(ctx, &assistpb.CreateAssistantMessageRequest{
 		Message: &assistpb.AssistantMessage{
-			Type:        string(msgType), //string(MessageKindUserMessage),
-			Payload:     msgPayload,      // TODO(jakule): Sanitize the payload
+			Type:        string(MessageKindUserMessage),
+			Payload:     userInput, // TODO(jakule): Sanitize the payload
```

Comment:
That's why you add your name next to `TODO`. Otherwise the wrong person will be `blame` for it :sweat_smile: